pyrightconfig.json

```diff
@@ -1,8 +1,8 @@
 {
   "include": [
-    "ambersim"
+    "ambersim",
   ],
-  "reportMissingImports": false,
+  "reportMissingImports": true,
   "reportMissingTypeStubs": false,
-  "reportGeneralTypeIssues": false,
+  "reportGeneralTypeIssues": true,
```

Comment:
`reportGeneralTypeIssues`: this is the main setting change that caused all these "errors" to pop up that I fixed.